### PR TITLE
Autodetect endianness on export

### LINKED ISSUE #15: control reverse bytes on export
Playing with tft for arduino, this example load raw images on screen.

http://www.rinkydinkelectronics.com/library.php?id=53

I was curious about raw format and not all raw viewer was able to show them.

I dig a bit into formats and ra viewer support (by far) lot more formats than any other raw viewer.

I tried to use raviewer as thumbailer and while on GUI export just works, from CLI looks like endianess (I guess reversed bytes are about that) is not parsed for input files

This is the expected result (what I get from GUI):

pic803 raw correct

but this is what I get from CLI:

    raviewer -f pic803.raw -e .

pic803 raw

Here is the original file

[pic803.zip](https://github.com/user-attachments/files/17612121/pic803.zip)

### PR DESCRIPTION
**Pull request by albfan**
fixes #15

### CODE REVIEW

**Comment by glatosinski** [comment]
Hello,

Thank you for your contribution. Ability to configure endianness during export is indeed a useful feature.

First things first, I'd fix the formatting of the code so that the CI will pass - there is one issue reported by `yapf`.

You can run code formatting with:

```
./.github/check_code_formatting.sh
```

File: raviewer/__main__.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def run(file_path, width, height, color_format, export, args):
             color_format, _ = classify_top1(img)
         if width == 0:
             width, _ = predict_resolution(img, color_format)[0]
-        img = parse_image(img.data_buffer, color_format, width)
```
Comment (glatosinski) [inline raviewer/__main__.py:69]: I'd rather not leave the endianness decision to the predictor mechanism by default - it is a heuristic that determines endianness based on certain patterns in the data, which may not always yield desired results. I'd prefer to stick to little endian as default.

A better approach would be to introduce a flag to arguments, e.g. `--endianness`, allowing to select between `little`, `big` and `auto`/`detect`, and leave the current default as is.

**Review by glatosinski — CHANGES_REQUESTED** [review]

File: raviewer/__main__.py (hunk 1 above)
Comment (albfan) [reply raviewer/__main__.py:69]: Of course, will rework it then

**Comment by glatosinski** [comment]
I introduced handling `--endianness` in GUI, thanks, merged

**Pull request merged**